chain/network/src/peer_manager/connection_store/mod.rs

```diff
@@ -0,0 +1,114 @@
+use crate::peer_manager::network_state::NetworkState;
+use crate::store;
+use crate::time;
+use crate::types::{ConnectionInfo, PeerType};
+use near_primitives::network::PeerId;
+use parking_lot::Mutex;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+/// The maximum number of stored outbound connections.
+pub const OUTBOUND_CONNECTIONS_LIMIT: usize = 40;
+/// How long a connection should survive before being stored.
+pub(crate) const STORED_CONNECTIONS_MIN_DURATION: time::Duration = time::Duration::minutes(10);
+
+struct Inner {
+    store: store::Store,
+    outbound: Vec<ConnectionInfo>,
+}
+
+impl Inner {
+    fn remove_outbound_connection(&mut self, peer_id: &PeerId) {
+        self.outbound.retain(|c| c.peer_info.id != *peer_id);
+        if let Err(err) = self.store.set_recent_outbound_connections(&self.outbound) {
+            tracing::error!(target: "network", ?err, "Failed to save recent outbound connections");
+        }
+    }
+
+    fn contains_outbound(&self, peer_id: &PeerId) -> bool {
+        for stored_info in &self.outbound {
+            if stored_info.peer_info.id == *peer_id {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    fn update(&mut self, clock: &time::Clock, state: Arc<NetworkState>) {
+        let now = clock.now();
+        let now_utc = clock.now_utc();
```

Comment:
this implementation may miss the fact that some reconnection happened (and therefore incorrectly consider a connection to be continuous). Would it be a lot of work to implement it more precisely? I.e. distinguish connections to the same peer?

Reply:
It was fortunately a very simple change